Sources/Testing/Issues/Confirmation.swift

```diff
@@ -96,13 +96,92 @@ public func confirmation<R>(
+) async rethrows -> R {
+  try await confirmation(
+    comment,
+    expectedCount: expectedCount ... expectedCount,
+    sourceLocation: sourceLocation,
+    body
+  )
+}
+
+/// Confirm that some event occurs during the invocation of a function.
+///
+/// - Parameters:
+///   - comment: An optional comment to apply to any issues generated by this
+///     function.
+///   - expectedCount: A range of integers indicating the number of times the
+///   	expected event should occur when `body` is invoked.
+///   - sourceLocation: The source location to which any recorded issues should
+///     be attributed.
+///   - body: The function to invoke.
+///
+/// - Returns: Whatever is returned by `body`.
+///
+/// - Throws: Whatever is thrown by `body`.
+///
+/// Use confirmations to check that an event occurs while a test is running in
+/// complex scenarios where `#expect()` and `#require()` are insufficient. For
+/// example, a confirmation may be useful when an expected event occurs:
+///
+/// - In a context that cannot be awaited by the calling function such as an
+///   event handler or delegate callback;
+/// - More than once, or never; or
+/// - As a callback that is invoked as part of a larger operation.
+///
+/// To use a confirmation, pass a closure containing the work to be performed.
+/// The testing library will then pass an instance of ``Confirmation`` to the
+/// closure. Every time the event in question occurs, the closure should call
+/// the confirmation:
+///
+/// ```swift
+/// let minBuns = 5
+/// let maxBuns = 10
+/// await confirmation(
+/// 	"Baked between \(minBuns) and \(maxBuns) buns",
+///   expectedCount: minBuns ... maxBuns
+/// ) { bunBaked in
+///   foodTruck.eventHandler = { event in
+///     if event == .baked(.cinnamonBun) {
+///       bunBaked()
+///     }
+///   }
+///   await foodTruck.bakeTray(of: .cinnamonBun)
+/// }
+/// ```
+///
+/// When the closure returns, the testing library checks if the confirmation's
+/// preconditions have been met, and records an issue if they have not.
+///
+/// If an exact count is expected, use
+/// ``confirmation(_:expectedCount:sourceLocation:_:)-7kfko`` instead.
+@_spi(Experimental)
+public func confirmation<R>(
+  _ comment: Comment? = nil,
+  expectedCount: some RangeExpression<Int>,
+  sourceLocation: SourceLocation = #_sourceLocation,
+  _ body: (Confirmation) async throws -> R
 ) async rethrows -> R {
   let confirmation = Confirmation()
   defer {
     let actualCount = confirmation.count.rawValue
-    if actualCount != expectedCount {
+    if !expectedCount.contains(actualCount) {
+      var comment = comment
+      let issueKind: Issue.Kind
+      if let expectedCount = expectedCount as? ClosedRange<Int>,
+         expectedCount.lowerBound == expectedCount.upperBound {
+        issueKind = .confirmationMiscounted(actual: actualCount, expected: expectedCount.lowerBound)
+      } else {
+        // TODO: define an issue kind for out-of-range confirmation failures
```

Comment:
Tried to help add it in #515.

Reply:
Much appreciated—as discussed off of GitHub, we'll hold off on a resolution here until we have a bit more team buy-in and know how best to stage in a change (as I'd like to actually _replace `.confirmationMiscounted` here.)

Reply:
I also think we should consolidate the new `.confirmationMiscounted` enum case here — can we do that, or file an issue to track it?